includes/Connector.class.php

```diff
@@ -358,4 +362,20 @@ protected function throwRequestException($message) {
 
 		throw $requestException;
 	}
+
+	protected function checkForRequestErrors($request, $response) {
+		// if curl timed out
+		if (curl_errno($request) && (string)curl_errno($request) === '28') {

```

Comment:
Would we care `if curl_error($request) && (string)curl_errno($request) !== '28'`? Because we don't capture that scenario.

Reply:
Yes, we probably would. I'll use the generic `RequestException` here

Reply:
@bartboy011 
@mthommes brings in a good point. We should handle timeouts, 4xx and 5xx as well as other curl errors.